src/traces/parcoords/defaults.js

```diff
@@ -55,6 +55,19 @@ function dimensionDefaults(dimensionIn, dimensionOut, parentOut, opts) {
         coerce('ticktext');
         coerce('tickformat');
         var range = coerce('range');
+        if(!range) {
```

Comment:
Hmm I would move this down to the plot step. That way we don't have to loop over the data during the defaults.

Reply:
Good call.
Done in https://github.com/plotly/plotly.js/pull/4011/commits/bf02101c5d8c51e2db07065f1a7ab20ce0e8ff55.